pymc/sampling.py

```diff
@@ -2093,6 +2094,68 @@ def sample_prior_predictive(
     return pm.to_inference_data(prior=prior, **ikwargs)
 
 
+def draw(vars, draws=500, mode=None, **kwargs) -> Dict[str, np.ndarray]:
+    """Draw samples for one variable or a list of variables
+
+    Parameters
+    ----------
+    vars : Iterable[TensorVariable]
+        A variable or a list of variables for which to draw samples.
+    draws : int
+        Number of samples needed to draw. Detaults to 500.
+    mode:
+        The mode used by ``aesara.function`` to compile the graph.
+    kwargs:
+        Keyword arguments for :func:`pymc.aesara.compile_pymc`
+
+    Returns
+    -------
+    Dict[str, np.ndarray]
+        A dictionary with variable names as keys and drawn samples as numpy arrays.
+
+    Examples
+    --------
+    .. code-block:: python
+        # Draw samples for one variable
+        with pm.Model():
+            x = pm.Normal("x")
+        x_draws = pm.draw(x)
+        assert x_draws["x"].shape == (500,)
+
+        # Draw 1000 samples for several variables
+        with pm.Model():
+            x = pm.Normal("x")
+            y = pm.Normal("y", shape=10)
+            z = pm.Uniform("z", shape=5)
+
+        num_draws = 1000
+        draws = pm.draw([x, y, z], draws=num_draws)
+        assert draws["x"].shape == (num_draws,)
+        assert draws["y"].shape == (num_draws, 10)
+        assert draws["z"].shape == (num_draws, 5)
+    """
+
+    if vars is None:
+        raise AssertionError("Must include at least one variable")
+
+    if isinstance(vars, tuple):
+        vars = list(vars)
+    elif not isinstance(vars, list):
+        vars = [vars]
```

Comment:
Actually we don't even need this. The outputs do not need to be in a list when we have a single variable

Reply:
I have test this code.
```
    if not isinstance(vars, (list, tuple)):
        vars = [vars]
```
It will fail to the following test if `vars` is a tuple.
```
        with pm.Model() as model:
            x = pm.Normal("x")
            y = pm.Normal("y", shape=10)
            z = pm.Uniform("z", shape=5)
        # Draw samples of a tuple variables
        draws = pm.draw((x, y, z), draws=num_draws)
        assert draws["x"].shape == (num_draws,)
        assert draws["y"].shape == (num_draws, 10)
        assert draws["z"].shape == (num_draws, 5)
```
This due to the pymc.aesara.compile_pymc(inputs, outputs) function expect outputs as a list.
`output_to_list = outputs if isinstance(outputs, list) else [outputs]`

```
ouputs = (x, y, z)
[outputs] = [(x, y, z)] # Not what expected (a list)
```
So, to be save, I convert the tuple to list.

```
    if isinstance(vars, tuple):
        vars = list(vars)
```

Reply:
We should change that line in `compile_pymc` then to be 
```python
 output_to_list = outputs if isinstance(outputs, (list, tuple)) else [outputs]
```